File: closetalk_backend/internal/model/device.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type Device struct {
	ID          uuid.UUID `json:"id"`
	UserID      uuid.UUID `json:"user_id"`
	DeviceName  string    `json:"device_name"`
	DeviceType  string    `json:"device_type"`  // phone | tablet | desktop | web
	Platform    string    `json:"platform"`     // android | ios | windows | macos | linux | web
	PublicKey   string    `json:"public_key,omitempty"`
	PushToken   string    `json:"push_token,omitempty"`
	AppVersion  string    `json:"app_version,omitempty"`
	IsActive    bool      `json:"is_active"`
	LinkedAt    time.Time `json:"linked_at"`
	LastActive  time.Time `json:"last_active"`
}

type LinkDeviceRequest struct {
	DevicePubKey string `json:"device_pub_key"`
	DeviceName   string `json:"device_name"`
	DeviceType   string `json:"device_type"`
	Platform     string `json:"platform"`
}

type LinkDeviceResponse struct {
	DeviceToken string `json:"device_token"`
	DeviceID    string `json:"device_id"`
}

type RevokeDeviceRequest struct {
	DeviceID string `json:"device_id"`
}

type DeviceResponse struct {
	ID          uuid.UUID `json:"id"`
	DeviceName  string    `json:"device_name"`
	DeviceType  string    `json:"device_type"`
	Platform    string    `json:"platform"`
	IsActive    bool      `json:"is_active"`
	LinkedAt    time.Time `json:"linked_at"`
	LastActive  time.Time `json:"last_active"`
}

File: closetalk_app/lib/models/device.dart
class Device {
  final String id;
  final String deviceName;
  final String deviceType;
  final String platform;
  final bool isActive;
  final DateTime linkedAt;
  final DateTime lastActive;

  Device({
    required this.id,
    required this.deviceName,
    required this.deviceType,
    required this.platform,
    required this.isActive,
    required this.linkedAt,
    required this.lastActive,
  });

  factory Device.fromJson(Map<String, dynamic> json) {
    return Device(
      id: json['id'] as String,
      deviceName: json['device_name'] as String,
      deviceType: json['device_type'] as String,
      platform: json['platform'] as String,
      isActive: json['is_active'] as bool? ?? true,
      linkedAt: DateTime.parse(json['linked_at'] as String),
      lastActive: DateTime.parse(json['last_active'] as String),
    );
  }
}

class LinkDeviceResponse {
  final String deviceToken;
  final String deviceId;

  LinkDeviceResponse({
    required this.deviceToken,
    required this.deviceId,
  });

  factory LinkDeviceResponse.fromJson(Map<String, dynamic> json) {
    return LinkDeviceResponse(
      deviceToken: json['device_token'] as String,
      deviceId: json['device_id'] as String,
    );
  }
}
